### PR TITLE
feat(redirect)[OK-292] Redirect to subpage from saved path

### PR DESCRIPTION
**Pull request by adbista**

### CODE REVIEW

File: client/src/lib/urlUtils.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { sanitizeUrl } from "@braintree/sanitize-url";
+import Cookies from "js-cookie";
+
+export const sanitize = (input: string): string => {
+  return sanitizeUrl(input);
+};
```
Comment (Strattoss) [inline client/src/lib/urlUtils.ts]: I'm not sure if this function is needed. We can just use sanitizeUrl directly instead :)

File: client/src/middleware.ts [hunk 2]
```diff
@@ -17,19 +17,26 @@ function isAuthenticated(request: NextRequest): boolean {
...
-  const { pathname } = request.nextUrl;
+  const { pathname, search } = request.nextUrl;
   if (isAuthenticated(request)) {
     if (pathname.startsWith("/auth")) {
       return NextResponse.redirect(new URL("/user-boards", request.url));
     }
   } else {
-    if ((pathname.startsWith("/user-boards") || pathname.startsWith("/board")) && !pathname.startsWith("/auth")) {
-      return NextResponse.redirect(new URL("/auth/login", request.url));
+    if (pathname.startsWith("/user-boards") && !pathname.startsWith("/auth")) {
```
Comment (Strattoss) [inline client/src/middleware.ts]: If `pathname` starts with `/user-boards`, can it start with `/auth` at the same time?

**Review by Strattoss — CHANGES_REQUESTED** [review]

File: client/src/lib/urlUtils.ts (hunk 1 above)
Comment (adbista) [reply client/src/lib/urlUtils.ts]: done

File: client/src/middleware.ts (hunk 2 above)
Comment (adbista) [reply client/src/middleware.ts]: oh yes, done

**Review by Strattoss — APPROVED** [review]

**Pull request merged**